src/Compilers/Core/Benchmarks/CreateSourceTextFromDisk.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Text;
+using BenchmarkDotNet.Attributes;
+using Microsoft.CodeAnalysis.Text;
+
+#nullable disable
+
+[MemoryDiagnoser]
+public class CreateSourceTextFromDisk
+{
+    public static string SmallFilePath;
+    public static string MediumFilePath;
+    public static string LargeFilePath;
+
+    [GlobalSetup]
+    public void GlobalSetup()
+    {
+        var dir = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+        Directory.CreateDirectory(dir);
+        SmallFilePath = writeCode("small", 1000);
+        MediumFilePath = writeCode("medium", 4000);
+        LargeFilePath = writeCode("medium", 8000);
+
+        string writeCode(string fileName, int count)
+        {
+            var code = getCode(count);
+            var filePath = Path.Combine(dir, fileName);
+            File.WriteAllText(filePath, code, Encoding.UTF8);
+            return filePath;
+        }
+
+        string getCode(int count)
+        {
+            var builder = new StringBuilder();
+            builder.AppendLine("/*");
+            var line = new string('=', 10);
+            for (int i = 0; i < count / 10; i++)
+            {
+                builder.AppendLine(line);
+            }
+            builder.AppendLine("*/");
+            return builder.ToString();
+        }
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public SourceText LoadFromFile(string filePath)
+    {
+        using var fileStream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1);
```

Comment:
>using var fileStream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1); [](http://example.com/codeflow?start=8&length=111)

What about putting the filestream creation/disposal in IterationSetup/IterationCleanup to make the benchmark results not include those?

Reply:
Does that generally have meaningful impact on the benchmarks? Happy to try this but also want to learn why you thought this was a good suggestion.

Reply:
The times you are measuring are in microseconds, right? That's fast enough that it seems like the FileStream creation/disposal might account for a significant portion.